halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/SecurityConfig.java

```diff
@@ -17,17 +17,32 @@
 
 package com.netflix.spinnaker.halyard.deploy.spinnaker.v1.profile;
 
+import com.netflix.spinnaker.halyard.deploy.spinnaker.v1.service.ServiceSettings;
 import lombok.Data;
 
 @Data
-public class SslConfig {
-  boolean enabled = false;
-  String keyAlias;
-  String keyStore;
-  String keyStoreType;
-  String keyStorePassword;
-  String trustStore;
-  String trustStoreType;
-  String trustStorePassword;
-  String clientAuth = "need";
+public class SecurityConfig {
+  BasicSecurity basic = new BasicSecurity();
+  User user = new User();
+
+  SecurityConfig(ServiceSettings settings) {
+    String username = settings.getUsername();
+    String password = settings.getPassword();
+    if (username != null && password != null) {
```

Comment:
these are generated by halyard, so there's no need to validate here. I'll tidy this up though to make it clear/less error prone if halyard omits a value in the future